ipaplatform/base/services.py

```diff
@@ -50,12 +50,15 @@
 # *some* naming, set them as in Red Hat distributions. Actual implementation
 # should make them available through knownservices.<name> and take care of
 # re-mapping internally, if needed
-wellknownservices = ['certmonger', 'dirsrv', 'httpd', 'ipa', 'krb5kdc',
-                     'dbus', 'nslcd', 'nscd', 'ntpd', 'portmap',
-                     'rpcbind', 'kadmin', 'sshd', 'autofs', 'rpcgssd',
-                     'rpcidmapd', 'pki_tomcatd', 'chronyd', 'domainname',
-                     'named', 'ods_enforcerd', 'ods_signerd', 'gssproxy',
-                     'nfs-utils', 'sssd', 'NetworkManager']
+wellknownservices = [
+    'certmonger', 'dirsrv', 'httpd', 'ipa', 'krb5kdc',
+    'dbus', 'nslcd', 'nscd', 'ntpd', 'portmap',
+    'rpcbind', 'kadmin', 'sshd', 'autofs', 'rpcgssd',
+    'rpcidmapd', 'pki_tomcatd', 'chronyd', 'domainname',
+    'named', 'ods_enforcerd', 'ods_signerd', 'gssproxy',
+    'nfs-utils', 'sssd', 'NetworkManager', 'ipa-custodia',
+    'ipa-dnskeysyncd', 'ipa-otpd', 'ipa-ods-exporter'
```

Comment:
I gave you a bad advice. The wellknownservices list can only contain ... drum role for a big surprise ... services. ``redhat_system_units`` initialization routine hard-codes ``.service`` suffix. Please remove ``ipa-ods-exporter`` again. It's a socket.

Reply:
Resolved: both ipa-ods-exporter.{socket,service} exist.